roles/metrics-export.lua

```diff
@@ -170,14 +168,48 @@ local function validate_http_endpoint(endpoint)
     end
 end
 
+-- check_server_httpd_role validates that httpd configuration and provided name exists.
+local function check_server_httpd_role(server)
+    local httpd_roles_cfg = require("config"):get("roles_cfg")['roles.httpd']
+    if httpd_roles_cfg == nil then
+        error("there is no configuration for httpd role")
+    end
+    if httpd_roles_cfg[server] == nil then
+        error(('server with name %s not found in httpd role config'):format(server))
+    end
+end
+
+-- get_server_from_httpd_role_cfg returns http server by provided server_name from httpd role.
+-- If server_name not found returns server by DEFAULT_SERVER_NAME.
+local function get_server_from_httpd_role_cfg(server_name)
+    local httpd_roles_cfg = require("config"):get("roles_cfg")['roles.httpd']
+    return httpd_roles_cfg[server_name] or
+           httpd_roles_cfg[httpd_role.DEFAULT_SERVER_NAME]
+end
+
+
 local function validate_http_node(node)
     if type(node) ~= "table" then
         error("http configuration node must be a table, got " .. type(node), 3)
     end
 
-    local _, _, err = parse_listen(node.listen)
-    if err ~= nil then
-        error("failed to parse http 'listen' param: " .. err, 3)
+    if node.server ~= nil then
+        if type(node.server) ~= 'string' then
+            error("server configuration sould be a string, got " .. type(node.server))
+        end
+
+        if node.listen ~= nil then
+            error("it is not possible to provide 'server' and 'listen' blocks simultaneously")
```

Comment:
Please, update the error level for other errors.
```suggestion
            error("server configuration sould be a string, got " .. type(node.server), 3)
        end

        if node.listen ~= nil then
            error("it is not possible to provide 'server' and 'listen' blocks simultaneously", 3)
```

Reply:
Updated.